src/storage/storage.hpp

```diff
@@ -178,13 +227,41 @@ class Storage : public aos::sm::launcher::StorageItf,
         }
     };
 
+    struct CertInfo {
+        uint8_t  mIssuer[aos::crypto::cCertIssuerSize];
+        size_t   mIssuerLen;
+        uint8_t  mSerial[aos::crypto::cCertSerialNumberSize];
+        size_t   mSerialLen;
+        char     mCertURL[aos::cURLLen];
+        char     mKeyURL[aos::cURLLen];
+        char     mCertType[aos::iam::certhandler::cCertTypeLen];
+        uint64_t mNotAfter;
+
+        /**
+         * Compares cert info.
+         *
+         * @param cert info to compare.
+         * @return bool.
+         */
+        bool operator==(const CertInfo& rhs) const
+        {
+            return memcmp(mIssuer, rhs.mIssuer, sizeof(mIssuer)) == 0
```

Comment:
first you should compare sizes and then do memcmp for size only.

Reply:
Done